frontend/src/components/datasets/versions/DatasetVersions.tsx

```diff
@@ -0,0 +1,243 @@
+	const [limit] = useState<number>(config.defaultVersionPerPage);
+	const [skip, setSkip] = useState<number>(0);
+	const [selectedDatasetVersion, setSelectedDatasetVersion] =
+		useState<DatasetFreezeOut>();
+	const [deleteDatasetVersionConfirmOpen, setDeleteDatasetVersionConfirmOpen] =
+		useState(false);
+
+	const getFreezeDatasets = (
+		datasetId: string | undefined,
+		skip: number,
+		limit: number
+	) => dispatch(getFreezeDatasetsAction(datasetId, skip, limit));
+
+	const deleteFreezeDataset = (
+		datasetId: string | undefined,
+		frozenVersionNum: number
+	) => {
+		dispatch(deleteFreezeDatasetAction(datasetId, frozenVersionNum));
+	};
+
+	const latestFrozenVersionNum = useSelector(
+		(state: RootState) => state.dataset.latestFrozenVersionNum
+	);
+	const deletedFrozenDataset = useSelector(
+		(state: RootState) => state.dataset.deletedFrozenDataset
+	);
+	const frozenDatasets = useSelector(
+		(state: RootState) => state.dataset.frozenDatasets.data
+	);
+	const pageMetadata = useSelector(
+		(state: RootState) => state.dataset.frozenDatasets.metadata
+	);
+
+	const history = useNavigate();
+
+	useEffect(() => {
+		if (currVersionPageNumParam) {
+			const pageNum = parseInt(currVersionPageNumParam, 10);
+			setCurrVersionPageNum(pageNum);
+			setSkip((pageNum - 1) * limit);
+		}
+	}, [currVersionPageNumParam]);
+
+	useEffect(() => {
+		let datasetId;
+		if (currDataset.origin_id) datasetId = currDataset.origin_id;
+		else datasetId = currDataset.id;
+		if (datasetId) getFreezeDatasets(datasetId, skip, limit);
+	}, [currDataset, deletedFrozenDataset, latestFrozenVersionNum, skip, limit]);
+
+	const handleVersionChange = (selectedDatasetId: string) => {
+		// current version flip to page 1
+		if (selectedDatasetId === currDataset.origin_id) {
+			history(`/datasets/${selectedDatasetId}?currVersionPageNum=1`);
+			// other version flip to the current page
+		} else {
+			history(
+				`/datasets/${selectedDatasetId}?currVersionPageNum=${currVersionPageNum}`
+			);
+		}
+	};
+
+	const handlePageChange = (_: ChangeEvent<unknown>, value: number) => {
+		const originDatasetId = currDataset.origin_id;
+		const newSkip = (value - 1) * limit;
+		setCurrVersionPageNum(value);
+		setSkip(newSkip);
+		getFreezeDatasets(originDatasetId, newSkip, limit);
+	};
+
+	// delete dataset version
+	const deleteSelectedDatasetVersion = () => {
+		if (
+			selectedDatasetVersion &&
+			selectedDatasetVersion.origin_id &&
+			selectedDatasetVersion.frozen_version_num
+		)
+			deleteFreezeDataset(
+				selectedDatasetVersion.origin_id,
+				selectedDatasetVersion.frozen_version_num
+			);
+		// after delete go to the current version
+		history(`/datasets/${currDataset.origin_id}?currVersionPageNum=1`);
+		setDeleteDatasetVersionConfirmOpen(false);
+	};
+
+	return (
+		<>
+			<ActionModal
+				actionOpen={deleteDatasetVersionConfirmOpen}
+				actionTitle="Delete Dataset"
+				actionText="Are you sure you want to delete this version? This process cannot be undone.
```

Comment:
I was wondering if we can have this text in some kind of config file. So that it's easier to change the text in a single place. No need to address this here. We can think about this later.

Reply:
Created an issue for that: https://github.com/clowder-framework/clowder2/issues/1100